src/ducks/transactions/TransactionModal.jsx

```diff
@@ -139,30 +142,59 @@ export const showAlertAfterApplicationDateUpdate = (transaction, t, f) => {
   )
 }
 
-const RecurrenceRow = ({ transaction, onClick }) => {
+const RecurrenceRow = withRouter(({ transaction, onClick, router }) => {
   const recurrence = transaction.recurrence && transaction.recurrence.data
   const { t } = useI18n()
+
+  const recurrenceRoute = recurrence ? `/recurrence/${recurrence._id}` : null
+  const handleClickHistory = ev => {
+    ev.preventDefault()
+    router.push(recurrenceRoute)
+  }
+
   return (
-    <TransactionModalRowMedia onClick={onClick}>
+    <TransactionModalRowMedia
+      align={recurrence ? 'top' : undefined}
+      onClick={onClick}
+    >
       <Img>
         <TransactionModalRowIcon icon={iconRecurrence} />
       </Img>
       <Bd>
-        {recurrence ? (
-          <>
-            ( prettyLabel(recurrence.label) ) <br />
-            <Caption>Every {recurrence.stats.median}</Caption>
-          </>
-        ) : (
-          <i>{t('Recurrence.not-recurring')}</i>
-        )}
+        <div>
+          {recurrence ? (
+            prettyLabel(recurrence.label)
+          ) : (
+            <i>{t('Recurrence.not-recurring')}</i>
+          )}
+          {recurrence ? (
+            <>
+              <br />
+              <Caption>
+                {t('Recurrence.frequency', {
+                  frequency: Math.floor(recurrence.stats.deltas.mean)
+                })}
+              </Caption>
+              {router.location.pathname !== recurrenceRoute ? (
+                <Chip
+                  onClick={handleClickHistory}
```

Comment:
nit: I don't know how often we'll click on it and how the user will use this feature, but when I can't cmd + click on something it annoys me

Reply:
Very good point. I'll use a link.